### PR TITLE
mrc-1838 friendlier error pages

### PR DESCRIPTION
**Pull request by hillalex**
Adds "page not found" and "something went wrong" erorr pages. Moves the top navbar which has no dynamic content out of the vue component and into the freemarker templates.

### CODE REVIEW

File: src/app/static/webpack.config.js [hunk 1]
```diff
@@ -69,34 +78,20 @@ const commonConfig = {
...
+        new VueLoaderPlugin(),
+        new HtmlWebpackPlugin({
+            hash: true,
+            template: 'public/index.ftl',
+            filename: 'index.ftl'
+        })
     ]
 };
 
-const appConfig = {...commonConfig,
```
Comment (hillalex) [inline src/app/static/webpack.config.js:76]: Also just noticed that this file could be simplified, I think must have originated from a copy paste from another webpack config with multiple configs exported.

File: src/app/src/main/kotlin/org/imperial/mrc/mint/controllers/MINTErrorController.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+class MINTErrorController(private val appProperties: AppProperties) : ErrorController {
+
+    override fun getErrorPath(): String {
+        return "/error"
+    }
+
+    @RequestMapping("/error")
+    fun handleError(request: HttpServletRequest): ModelAndView {
+        val status = request.getAttribute(RequestDispatcher.ERROR_STATUS_CODE)
+                ?.toString()?.toInt()?: 0
```
Comment (EmmaLRussell) [inline src/app/src/main/kotlin/org/imperial/mrc/mint/controllers/MINTErrorController.kt:24]: Codecov is complaining about not testing the null case here I think

File: src/app/src/test/kotlin/org/imperial/mrc/mint/integration/endpoints/ErrorTests.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package org.imperial.mrc.mint.integration.endpoints
+
+import org.assertj.core.api.AssertionsForClassTypes.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.web.client.getForEntity
+
+class ErrorTests: EndpointTests() {
```
Comment (EmmaLRussell) [inline src/app/src/test/kotlin/org/imperial/mrc/mint/integration/endpoints/ErrorTests.kt:7]: I was going to say should probably  test the 500 case too, but then couldn't think of a way to force a 500: I modified the code to self-sabotage to test it manually. Could maybe write a test with a self-defeating config, e.g. wrong mintr endpoint, I guess that would probably throw an exception?

File: src/app/static/templates/500.ftl [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<#include "partials/header.ftl">
+<div class="container mt-5">
+    <div class="row">
+        <div class="col-lg-6 offset-lg-3 col-10 offset-1">
+            <h2>Something Went Wrong</h2>
```
Comment (EmmaLRussell) [inline src/app/static/templates/500.ftl:5]: Maybe we should, in future tickets, log the error and provide an id here as we do in HINT/VIMC

File: src/app/src/main/kotlin/org/imperial/mrc/mint/controllers/MINTErrorController.kt (hunk 2 above)
Comment (hillalex) [reply src/app/src/main/kotlin/org/imperial/mrc/mint/controllers/MINTErrorController.kt:24]: Yes, although I am testing the null case:
https://github.com/mrc-ide/mint/pull/57/files#diff-102c4fb85bfb825bcb76912a6590c4148f709fd4992e095b757b5ae606f6a31eR61

 But it's not smart enough to work out that if the attribute is not null, then the subsequent string and int must also be not null.

File: src/app/src/test/kotlin/org/imperial/mrc/mint/integration/endpoints/ErrorTests.kt (hunk 3 above)
Comment (hillalex) [reply src/app/src/test/kotlin/org/imperial/mrc/mint/integration/endpoints/ErrorTests.kt:7]: I'm not too worried given the unit tests cover the 500 case?

Comment (hillalex) [reply src/app/src/test/kotlin/org/imperial/mrc/mint/integration/endpoints/ErrorTests.kt:7]: Although I suppose the "500.ftl" template could be missing in theory, ok I'll figure something out.

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**